### PR TITLE
23819 - Continuation Application Outputs Update

### PR DESCRIPTION
**Pull request by ArwenQin**
*Issue #:* /bcgov/entity#23819

*Description of changes:*
- Updated the Continuation Application Output per new design

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of the lear license (Apache 2.0).

### CODE REVIEW

File: legal-api/report-templates/template-parts/common/addresses.html [hunk 1]
```diff
@@ -1,6 +1,11 @@
 {% if business.legalType not in ['GP', 'SP'] %}
 <div class="no-page-break">
-   <div class="separator mt-4"></div>
+   {% if header.name == 'continuationIn' and not business.legalType in ['BEN', 'CBEN', 'CC', 'CCC', 'ULC', 'CUL'] 
+      and not nameRequest.nrNumber %}
+   <!-- Do not render the separator -->
+   {% else %}
+      <div class="separator mt-4"></div>
+   {% endif %}
```
Comment (ArwenQin) [inline legal-api/report-templates/template-parts/common/addresses.html:8]: Per new design, for numbered company, and without benefit company statement - we will not show the name and statement section. Thus address became the first section, so we need to hide the separator above the address
image

File: legal-api/src/legal_api/reports/report.py [hunk 2]
```diff
@@ -281,7 +283,7 @@ def _format_filing_json(self, filing):  # pylint: disable=too-many-branches, too
         elif self._report_key in ['certificateOfNameChange', 'certificateOfNameCorrection']:
             self._format_name_change_data(filing)
         elif self._report_key == 'certificateOfRestoration':
-            self._format_certificate_of_restoration_data(filing)
+            self._certificate_of_restorationformat__data(filing)
```
Comment (vysakh-menon-aot) [inline legal-api/src/legal_api/reports/report.py:286]: mistake?

**Comment by ketaki-deodhar** [comment]
@ArwenQin please fix the linting issue

File: legal-api/src/legal_api/reports/report.py (hunk 2 above)
Comment (ArwenQin) [reply legal-api/src/legal_api/reports/report.py:286]: Sorry it's by mistake. Will update

File: legal-api/src/legal_api/reports/report.py [hunk 3]
```diff
@@ -946,6 +948,11 @@ def _format_continuation_in_data(self, filing):
         # set expro business
         if filing['continuationIn'].get('business'):
             filing['exproBusiness'] = filing['continuationIn']['business']
+            # format founding date
+            founding_date_str = filing['exproBusiness']['foundingDate']
```
Comment (vysakh-menon-aot) [inline legal-api/src/legal_api/reports/report.py:952]: @severinbeauvais did we include this now in UI? We need to add validation if we are collecting this value

Comment (severinbeauvais) [reply legal-api/src/legal_api/reports/report.py:952]: Should be `['business']`, ie

image

File: legal-api/report-templates/continuationApplication.html [hunk 4]
```diff
@@ -24,29 +24,30 @@
...
-            {% else %}
-               {{ numberedDescription }}
-            {% endif %}
+               {% if business.legalName %}
+                  {{ business.legalName }}
+               {% elif nameRequest.legalName %}
+                  {{ nameRequest.legalName }}
+               {% else %}
+                  {{numberedDescription}}
+          {% endif %}
```
Comment (vysakh-menon-aot) [inline legal-api/report-templates/continuationApplication.html:33]: why do we need this change?

Comment (ArwenQin) [reply legal-api/report-templates/continuationApplication.html:33]: Current, it shows numbered company Description
image

Per new design, it shows the legal name if applicable:
image

After update:
image

File: legal-api/report-templates/template-parts/common/businessDetails.html [hunk 5]
```diff
@@ -327,9 +327,11 @@
...
             <span> | Registration #{{business.identifier}}</span>
         {% else %}
-           {% if header.name != 'incorporationApplication' %}
-             <!-- not required for incorporationApplication -->
-             <span> | Incorporation #{{business.identifier}}</span>
+            {% if header.name == 'continuationIn' %}
+               <span> | Continuation #{{business.identifier}}</span>
+            {% elif header.name != 'incorporationApplication' %}
+               <!-- not required for incorporationApplication -->
+               <span> | Incorporation #{{business.identifier}}</span>
```
Comment (vysakh-menon-aot) [inline legal-api/report-templates/template-parts/common/businessDetails.html:334]: As per the design this is not changed. Please confirm with Jacqueline
image

File: legal-api/report-templates/template-parts/continuation/companyNameStmt.html [hunk 6]
```diff
@@ -0,0 +1,25 @@
...
+                {% for translation in listOfTranslations %}
+                    <tr>
+                        <td class="pt-2">&bull;</td>
+                        <td class="pt-2">{{translation.name|upper}}</td>
+                    </tr>
+                {% endfor %}
+            </table>
+        </div>
+    {% endif %}
+</div>
```
Comment (vysakh-menon-aot) [inline legal-api/report-templates/template-parts/continuation/companyNameStmt.html:24]: This need to be outside of `{% if nameRequest.nrNumber%}`. A business can have name translation without NR

File: legal-api/src/legal_api/reports/report.py (hunk 3 above)
Comment (vysakh-menon-aot) [reply legal-api/src/legal_api/reports/report.py:952]: > is it ok to use 'exproBusiness' because line 950 set 'exproBusiness'? image

Yes. I was asking about foundingDate. we don't have a validation for that.

File: legal-api/report-templates/template-parts/common/businessDetails.html (hunk 5 above)
Comment (ArwenQin) [reply legal-api/report-templates/template-parts/common/businessDetails.html:334]: Jacqueline said use "Incorporation", will change back.

File: legal-api/src/legal_api/reports/report.py (hunk 3 above)
Comment (severinbeauvais) [reply legal-api/src/legal_api/reports/report.py:952]: Then, yes, we include foundingDate in the UI for an expro, and _someone_ needs to add Legal API validation for that.

What should be checked?
- valid date/time format
- is today or before
- anything else?

Sample filing JSON:
image

Comment (vysakh-menon-aot) [reply legal-api/src/legal_api/reports/report.py:952]: we already validate legal name of expro by fetching from colin, can do the same thing for foundingDate

File: legal-api/report-templates/continuationApplication.html (hunk 4 above)
Comment (ArwenQin) [reply legal-api/report-templates/continuationApplication.html]: Updated per new design:
when the numbered company didn't get a legal name yet, it shows numbered company Description;
After a numbered company gets a legal name (9900775 B.C. LTD), or it has a name request, it shows the legal name

File: legal-api/report-templates/template-parts/continuation/companyNameStmt.html (hunk 6 above)
Comment (ArwenQin) [reply legal-api/report-templates/template-parts/continuation/companyNameStmt.html:28]: Updated. this is not needed per new design. We will always show the name statement section

File: legal-api/report-templates/continuationApplication.html [hunk 7]
```diff
@@ -24,29 +24,28 @@
...
-            {% if nameRequest.legalName %}
-               {{ nameRequest.legalName }}
-            {% else %}
-               {{ numberedDescription }}
-            {% endif %}
+               {% if business.legalName %}
+                  {{ business.legalName }}
+               {% else %}
+                  {{numberedDescription}}
+          {% endif %}
```
Comment (vysakh-menon-aot) [inline legal-api/report-templates/continuationApplication.html:31]: You still need this. 
Once business created you will have named/numbered legal name in `business.legalName`.
If business not create and it a named company `nameRequest.legalName`
If business not create and it a numbered company `numberedDescription`
```suggestion
               {% if business.legalName %}
                  {{ business.legalName }}
               {% elif nameRequest.legalName %}
                  {{ nameRequest.legalName }}
               {% else %}
                  {{numberedDescription}}
               {% endif %}
```

File: legal-api/report-templates/continuationApplication.html [hunk 8]
```diff
@@ -24,29 +24,28 @@
...
          [[continuation/exproRegistrationInBc.html]]
          [[continuation/authorization.html]]
-         [[incorporation-application/completingParty.html]]
       </div>
       <div class="section-header mt-7 pt-4">NOTICE OF ARTICLES</div>
-      <div class="container pt-4">
+      <div class="container pt-2">
          [[common/statement.html]]
-         [[incorporation-application/benefitCompanyStmt.html]]
+         [[continuation/companyNameStmt.html]]
```
Comment (vysakh-menon-aot) [inline legal-api/report-templates/continuationApplication.html:48]: since you revert the changes, we don't need a new file you can refer back to `[[incorporation-application/benefitCompanyStmt.html]]`

Comment (ArwenQin) [reply legal-api/report-templates/continuationApplication.html:48]: incorporation-application/benefitCompanyStmt.html will show 'A company name using the Incorporation Number will be provided..' for all numbered companies, even those already got the legal name.
image
it didn't check the business.legal name
Maybe we update incorporation-application/benefitCompanyStmt.html ?

File: legal-api/report-templates/continuationApplication.html (hunk 7 above)
Comment (ArwenQin) [reply legal-api/report-templates/continuationApplication.html:31]: Thanks, will update

File: legal-api/report-templates/continuationApplication.html (hunk 8 above)
Comment (ArwenQin) [reply legal-api/report-templates/continuationApplication.html]: Deleted the new file, refer back to [[incorporation-application/benefitCompanyStmt.html]]
And I updated [[incorporation-application/benefitCompanyStmt.html]] to check business.legal name

File: legal-api/src/legal_api/reports/report.py [hunk 9]
```diff
@@ -960,8 +966,8 @@ def _format_continuation_in_data(self, filing):
         region = None
         if region_code and region_code.upper() != 'FEDERAL':
             region = pycountry.subdivisions.get(code=f'{country_code}-{region_code}')
-            filing['foreignJurisdiction']['region'] = region.name.upper() if region_code else ''
-        filing['foreignJurisdiction']['country'] = country.name.upper()
+            filing['foreignJurisdiction']['region'] = region.name if region_code else ''
```
Comment (ArwenQin) [inline legal-api/src/legal_api/reports/report.py:969]: This is to match the design:
image

File: legal-api/report-templates/continuationApplication.html (hunk 7 above)
Comment (vysakh-menon-aot) [reply legal-api/report-templates/continuationApplication.html]: while updating this, can you update here as well https://github.com/bcgov/lear/blob/main/legal-api/report-templates/incorporationApplication.html#L27

Comment (ArwenQin) [reply legal-api/report-templates/continuationApplication.html]: Sure, just updated

**Review by vysakh-menon-aot — APPROVED** [review]

File: legal-api/src/legal_api/reports/report.py (hunk 3 above)
Comment (severinbeauvais) [reply legal-api/src/legal_api/reports/report.py:951]: What's the path forward for this -- do it as part of an existing ticket or create a new ticket?

File: legal-api/report-templates/continuationApplication.html
Comment (severinbeauvais) [inline legal-api/report-templates/continuationApplication.html:1]: These code changes look good. Do you have a sample output (esp of lines 40-48)?

File: legal-api/report-templates/template-parts/continuation/effectiveDate.html [hunk 10]
```diff
@@ -3,7 +3,7 @@
     <div class="section-title mt-4">Continuation Effective Date</div>
     <div class="section-data mt-4">
         {% if header.isFutureEffective %}
-            {{effective_date_time}}
+            The continuation is to take effect on {{effective_date_time}}.
```
Comment (severinbeauvais) [inline legal-api/report-templates/template-parts/continuation/effectiveDate.html:6]: Do you have a sample output?

(Is effective_date_time something like, "October 7, 2024 at 8:00 pm Pacific time"?)

Comment (ArwenQin) [reply legal-api/report-templates/template-parts/continuation/effectiveDate.html:6]: image

File: legal-api/src/legal_api/reports/report.py (hunk 3 above)
Comment (vysakh-menon-aot) [reply legal-api/src/legal_api/reports/report.py:951]: create new ticket. we dont have any pending ticket to work on

File: legal-api/report-templates/template-parts/continuation/foreignJurisdiction.html [hunk 11]
```diff
@@ -1,34 +1,33 @@
...
+                <div class="pt-2">{{foreignJurisdiction.identifier}}</div>
+                <div class="pt-6">
+                    <span class="section-sub-title">Previous Jurisdiction</span>
+                </div>
+                <div class="pt-2">
+                    {% if foreignJurisdiction.region %}
+                        {{foreignJurisdiction.region}}
+                    {% else %}
+                        {{foreignJurisdiction.country}}
+                    {% endif %}
```
Comment (severinbeauvais) [inline legal-api/report-templates/template-parts/continuation/foreignJurisdiction.html:29]: I think this isn't in line with what the UI is displaying. This may be correct, but can you please confirm with Jacqueline?

Ref: [UI code](https://github.com/bcgov/business-create-ui/blob/967cd80f8dd53ff6ef83fb52ff2ebdd6d70e3fa0/src/components/ContinuationIn/SummaryBusinessPreviousJurisdiction.vue#L237)

Sample strings in UI:
- Federal
- Alberta
- Canada (should never happen but it's a fallback in case region is invalid)
- California, US
- US (should never happen but it's a fallback in case region is invalid)
- France

File: legal-api/report-templates/continuationApplication.html
Comment (severinbeauvais) [reply legal-api/report-templates/continuationApplication.html:1]: Looks good! Resolving.

File: legal-api/report-templates/template-parts/continuation/effectiveDate.html (hunk 10 above)
Comment (severinbeauvais) [reply legal-api/report-templates/template-parts/continuation/effectiveDate.html:6]: Excellent, thanks. Resolving.

File: legal-api/src/legal_api/reports/report.py (hunk 3 above)
Comment (ArwenQin) [reply legal-api/src/legal_api/reports/report.py:951]: New ticket created #24073
https://app.zenhub.com/workspaces/entities---olga-65af15f59e89f5043c2911f7/issues/gh/bcgov/entity/24073

**Comment by ArwenQin** [comment]
New ticket created #24073
https://app.zenhub.com/workspaces/entities---olga-65af15f59e89f5043c2911f7/issues/gh/bcgov/entity/24073

**Comment by ArwenQin** [comment]
For the Previous Jurisdiction String:
Canadian: Alberta, Canada
United States: California, USA
Federal: Federal, Canada 
International: Japan

Jacqueline will create a new ticket to update them.
